clap_complete/src/dynamic/completer.rs

```diff
@@ -3,6 +3,91 @@ use std::ffi::OsString;
 
 use clap_lex::OsStrExt as _;
 
+/// Gets all the long options, their visible aliases and flags of a [`clap::Command`].
+/// Includes `help` and `version` depending on the [`clap::Command`] settings.
+pub fn longs_and_visible_aliases(p: &clap::Command) -> Vec<String> {
```

Comment:
- These utilities shouldn't be `pub`
- Please put utilities liek this at the bottom so the focus is on the top-level content